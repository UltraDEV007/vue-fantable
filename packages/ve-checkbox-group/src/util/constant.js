// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
