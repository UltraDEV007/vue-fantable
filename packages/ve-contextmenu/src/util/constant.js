// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
