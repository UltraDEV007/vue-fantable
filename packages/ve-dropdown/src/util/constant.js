// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
