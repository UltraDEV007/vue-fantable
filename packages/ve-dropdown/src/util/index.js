/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
