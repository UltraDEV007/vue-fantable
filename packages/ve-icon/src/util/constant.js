// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
