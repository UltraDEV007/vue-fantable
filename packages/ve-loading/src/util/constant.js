// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
