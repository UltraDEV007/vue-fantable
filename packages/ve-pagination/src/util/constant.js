// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
