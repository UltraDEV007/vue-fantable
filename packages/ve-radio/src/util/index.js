import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
