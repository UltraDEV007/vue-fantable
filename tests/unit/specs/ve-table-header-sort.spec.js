import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";

describe("veTable header sort", () => {
    const TABLE_DATA = [
        {
            name: "John",
            age: 25,
            weight: 66,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shanghai",
        },
        {
            name: "Dickerson",
            age: 20,
            weight: 70,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Beijing",
        },
        {
            name: "Larsen",
            age: 18,
            weight: 65,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Chongqing",
        },
        {
            name: "Geneva",
            age: 17,
            weight: 80,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Xiamen",
        },
        {
            name: "Jami",
            age: 26,
            weight: 72,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
        },
    ];

    it("render single field sort", () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const wrapper = mount({
            render() {
                return (
                    <veTable
                        sortOption={this.sortOption}
                        columns={this.columns}
                        tableData={this.tableData}
                        rowKeyFieldName="rowkey"
                    />
                );
            },
            data() {
                return {
                    columns: columns,
                    tableData: TABLE_DATA,
                    sortOption: {
                        sortChange: (params) => {},
                    },
                };
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render multiple field sort", () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const wrapper = mount({
            render() {
                return (
                    <veTable
                        sortOption={this.sortOption}
                        columns={this.columns}
                        tableData={this.tableData}
                        rowKeyFieldName="rowkey"
                    />
                );
            },
            data() {
                return {
                    columns: columns,
                    tableData: TABLE_DATA,
                    sortOption: {
                        multipleSort: true,
                        sortChange: (params) => {},
                    },
                };
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("single field sort", async () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
                fixed: "left", // combine with column fixed issue #454
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const mockFn = jest.fn();
        const callBackData = {
            age: "asc",
            weight: "",
        };

        const wrapper = mount(veTable, {
            propsData: {
                columns: columns,
                tableData: TABLE_DATA,
                sortOption: {
                    sortChange: (params) => {
                        mockFn(params);
                    },
                },
                rowKeyFieldName: "rowkey",
            },
        });

        const thEls = wrapper.findAll(".ve-table-header-tr th");
        expect(thEls.at(1).find(".ve-table-sort").exists()).toBe(true);

        expect(
            thEls
                .at(2)
                .findAll(".ve-table-sort .ve-table-sort-icon")
                .at(0)
                .find(".active")
                .exists(),
        ).toBe(true);

        thEls.at(1).find(".ve-table-sortable-column").trigger("click");

        await later();

        expect(mockFn).toBeCalled();
        expect(mockFn).toHaveBeenCalledWith(callBackData);

        // 只能单个字段排序
        expect(wrapper.vm.colgroups.filter((x) => x.sortBy).length).toEqual(1);

        expect(wrapper.vm.colgroups).toEqual([
            {
                _colspan: 1,
                _keys: "a",
                _level: 1,
                _realTimeWidth: undefined,
                _rowspan: 1,
                align: "left",
                field: "name",
                fixed: "left",
                key: "a",
                title: "Name",
            },
            {
                _colspan: 1,
                _keys: "b",
                _level: 1,
                _realTimeWidth: undefined,
                _rowspan: 1,
                align: "center",
                field: "age",
                key: "b",
                sortBy: "asc",
                title: "Age",
            },
            {
                _colspan: 1,
                _keys: "c",
                _level: 1,
                _realTimeWidth: undefined,
                _rowspan: 1,
                align: "center",
                field: "weight",
                key: "c",
                sortBy: "",
                title: "Weight(kg)",
            },
            {
                _colspan: 1,
                _keys: "d",
                _level: 1,
                _realTimeWidth: undefined,
                _rowspan: 1,
                align: "center",
                field: "hobby",
                key: "d",
                title: "Hobby",
            },
            {
                _colspan: 1,
                _keys: "e",
                _level: 1,
                _realTimeWidth: undefined,
                _rowspan: 1,
                align: "left",
                field: "address",
                key: "e",
                title: "Address",
            },
        ]);

        expect(
            thEls
                .at(1)
                .findAll(".ve-table-sort .ve-table-sort-icon")
                .at(0)
                .find(".active")
                .exists(),
        ).toBe(true);
    });

    it("multiple field sort", async () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const mockFn = jest.fn();
        const callBackData = {
            age: "asc",
            weight: "asc",
        };

        const wrapper = mount(veTable, {
            propsData: {
                columns: columns,
                tableData: TABLE_DATA,
                sortOption: {
                    multipleSort: true,
                    sortChange: (params) => {
                        mockFn(params);
                    },
                },
                rowKeyFieldName: "rowkey",
            },
        });

        const thEls = wrapper.findAll(".ve-table-header-tr th");
        expect(thEls.at(1).find(".ve-table-sort").exists()).toBe(true);

        expect(
            thEls
                .at(2)
                .findAll(".ve-table-sort .ve-table-sort-icon")
                .at(0)
                .find(".active")
                .exists(),
        ).toBe(true);

        thEls.at(1).find(".ve-table-sortable-column").trigger("click");

        await later();

        expect(mockFn).toBeCalled();
        expect(mockFn).toHaveBeenCalledWith(callBackData);

        expect(
            wrapper.findAll(".ve-table-sort .ve-table-sort-icon.active").length,
        ).toBe(2);

        // 多个字段排序
        expect(wrapper.vm.colgroups.filter((x) => x.sortBy).length).toEqual(2);
    });

    it("sort always", async () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const mockFn = jest.fn();

        const wrapper = mount(veTable, {
            propsData: {
                columns: columns,
                tableData: TABLE_DATA,
                sortOption: {
                    sortAlways: true,
                    sortChange: (params) => {
                        mockFn(params);
                    },
                },
                rowKeyFieldName: "rowkey",
            },
        });

        const thEls = wrapper.findAll(".ve-table-header-tr th");

        thEls.at(1).find(".ve-table-sortable-column").trigger("click");

        await later();

        expect(mockFn).toHaveBeenCalledWith({
            age: "asc",
            weight: "",
        });

        thEls.at(1).find(".ve-table-sortable-column").trigger("click");

        await later();

        expect(mockFn).toHaveBeenCalledWith({
            age: "desc",
            weight: "",
        });
    });
});
