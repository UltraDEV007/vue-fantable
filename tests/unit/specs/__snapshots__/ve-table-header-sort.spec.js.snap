// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veTable header sort render multiple field sort 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col>
            <col>
            <col>
            <col>
            <col>
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-sortable-column" style="text-align: center; top: 0px;">Age<span class="fan-table-sort"><i class="ve-icon iconfont-vet icon-vet-sort-top-arrow fan-table-sort-icon fan-table-sort-icon-top"></i><i class="ve-icon iconfont-vet icon-vet-sort-bottom-arrow fan-table-sort-icon fan-table-sort-icon-bottom"></i></span></th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-sortable-column" style="text-align: center; top: 0px;">Weight(kg)<span class="fan-table-sort"><i class="ve-icon iconfont-vet icon-vet-sort-top-arrow fan-table-sort-icon fan-table-sort-icon-top active"></i><i class="ve-icon iconfont-vet icon-vet-sort-bottom-arrow fan-table-sort-icon fan-table-sort-icon-bottom"></i></span></th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">John</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">25</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">66</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">70</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Larsen</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">18</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">65</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Geneva</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">17</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">80</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Jami</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">26</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">72</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;

exports[`veTable header sort render single field sort 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col>
            <col>
            <col>
            <col>
            <col>
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-sortable-column" style="text-align: center; top: 0px;">Age<span class="fan-table-sort"><i class="ve-icon iconfont-vet icon-vet-sort-top-arrow fan-table-sort-icon fan-table-sort-icon-top"></i><i class="ve-icon iconfont-vet icon-vet-sort-bottom-arrow fan-table-sort-icon fan-table-sort-icon-bottom"></i></span></th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-sortable-column" style="text-align: center; top: 0px;">Weight(kg)<span class="fan-table-sort"><i class="ve-icon iconfont-vet icon-vet-sort-top-arrow fan-table-sort-icon fan-table-sort-icon-top active"></i><i class="ve-icon iconfont-vet icon-vet-sort-bottom-arrow fan-table-sort-icon fan-table-sort-icon-bottom"></i></span></th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">John</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">25</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">66</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">70</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Larsen</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">18</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">65</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Geneva</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">17</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">80</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Jami</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: center;">26</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">72</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="e" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;
